cpp/arcticdb/stream/python_bindings.cpp

```diff
@@ -24,6 +24,14 @@ namespace py = pybind11;
 namespace arcticdb {
 using namespace arcticdb::python_util;
 
+std::vector<FieldRef> field_collection_to_ref_vector(const FieldCollection& fields){
+    auto result = std::vector<FieldRef>();
```

Comment:
1. You can reserve using `FieldCollection::size`
2. Given that `FieldCollection` I think we can use `std::copy`

Reply:
My personal preference is to use `vector::reserve` + `std::back_inserter` (instead of setting the size in the constructor of the vector) to avoid the call to the default constructor which will later be overridden anyways. I'm aware that it's a micro-optimization that probably won't change anything, so I'll leave it up to you to decide if you want to leave it as is.